### PR TITLE
fix the intermittent failure for testCleanResyncGhosts

### PR DESCRIPTION
**Pull request by helenren**

### CODE REVIEW

File: test/SFSmartSyncTestSuite.js [hunk 1]
```diff
@@ -2849,13 +2849,13 @@ SmartSyncTestSuite.prototype.testCleanResyncGhosts = function() {
     console.log("# In SmartSyncTestSuite.testCleanResyncGhosts");
     var self = this;
     var idToName = {};
-    var delRecordId;
+    var ids;
+    var delRecordId
```
Comment (bhariharan) [inline test/SFSmartSyncTestSuite.js]: Missing ```;```.

File: test/SFSmartSyncTestSuite.js [hunk 2]
```diff
@@ -2901,8 +2902,10 @@ SmartSyncTestSuite.prototype.testCleanResyncGhosts = function() {
             var secondEntry = entries[1];
             var firstId = firstEntry["Id"];
             var secondId = secondEntry["Id"];
-            QUnit.equals(firstId, firstExpectedId, "ID should not still exist in SmartStore");
-            QUnit.equals(secondId, secondExpectedId, "ID should not still exist in SmartStore");
+            Console.log(“## Actual records Ids are ” + firstId + “ and ” + secondId);
```
Comment (bhariharan) [inline test/SFSmartSyncTestSuite.js]: @helenren Should be lowercase ```console```, and the ```"``` look like the wrong ones I think.

**Comment by wmathurin** [comment]
The tests can be run in a browser. It's good way to make sure everything works without having to run xcode or visual studio. For smartstore and smartsync they use MockSmartStore.js and MockSmartSync.js.

You need to run a browser with same origin policy disabled. For example on a mac, you can do:
```
/Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome --disable-web-security --user-data-dir
```
Then open file:///path-to-shared-repo-clone/test/test.html

**Comment by helenren** [comment]
@wmathurin  thanks for your tips, let me have a verification.

**Comment by wmathurin** [comment]
By the way @bhariharan does MockSmartSync.js has a clean ghost method?

**Comment by wmathurin** [comment]
I checked and MockSmartSync.js doesn't have a mock implementation for cleanResyncGhosts so the test won't pass when running in the browser. We should create a small story for adding that mock method. It can probably wait after 4.2 (the most important is having the test pass on iOS and Android).

**Comment by wmathurin** [comment]
I wonder if we should have the tests run in a browser on the build machine as well.

**Comment by helenren** [comment]
So, the test in a browser most is target to test the server side, is that correct?

**Comment by wmathurin** [comment]
It's a bit painful developing inside a container (app need to be restarted every time etc). With hybrid app, one tries to go as far as possible inside the browser. We do that also with our hybrid sample apps and our hybrid tests. We develop them in the browser (the container functionality being provided by MockCordova, MockSmartStore etc) but login / network calls are real. Running those tests on the build machine would guarantee that our Mocks are up to date. We hope external developers can leverage them too (the Mocks).

**Comment by helenren** [comment]
got your point. Yes, we should add test via browser. Will create a story for it. For external dev to leverage those mock stuff, we have instruction in the release doc or other doc, right?

**Comment by bhariharan** [comment]
@wmathurin ```MockSmartSync.js``` does not have ```cleanReSyncGhosts```. It is probably lacking in some other areas too.

**Comment by helenren** [comment]
@wmathurin @bhariharan is there any other concern on this change? Actually this change majorly is to add log to help to understand this intermittent failure.

**Comment by wmathurin** [comment]
Looks fine to me.

**Comment by kchitalia** [comment]
LGTM

**Pull request merged**